### PR TITLE
API: harmonize drop/reindex/rename args (GH12392) - drop

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
xref #12392. 
For now this only deals with `drop` (making it similar with `reindex`), and not the other way around (`reindex` similar to `drop`). So the title is not fully correct, but this is the more easy part.

### CODE REVIEW

File: doc/source/whatsnew/v0.21.0.txt [hunk 1]
```diff
@@ -89,6 +89,24 @@ This does not raise any obvious exceptions, but also does not create a new colum
 
 Setting a list-like data structure into a new attribute now raise a ``UserWarning`` about the potential for unexpected behavior. See :ref:`Attribute Access <indexing.attribute_access>`.
 
+``drop`` now also accepts index/columns keywords
```
Comment (jreback) [inline doc/source/whatsnew/v0.21.0.txt]: add ref here

File: pandas/core/generic.py [hunk 2]
```diff
@@ -2354,36 +2361,62 @@ def drop(self, labels, axis=0, level=None, inplace=False, errors='raise'):
...
+            if labels is not None:
+                obj = obj._drop_axis(labels, axis, level=level, errors=errors)
+
+        if inplace:
+            self._update_inplace(obj)
+        else:
+            return obj
+
+    def _drop_axis(self, labels, axis, level=None, errors='raise'):
+
```
Comment (jreback) [inline pandas/core/generic.py]: add doc string here

File: pandas/tests/frame/test_axis_select_reindex.py [hunk 3]
```diff
@@ -146,6 +146,38 @@ def test_drop_multiindex_not_lexsorted(self):
...
+        tm.assert_frame_equal(res1, res2)
+
+        res1 = df.drop(['a'], axis=0)
+        res2 = df.drop(index=['a'])
+        tm.assert_frame_equal(res1, res2)
+
+        res1 = df.drop(['a'], axis=0).drop(['d'], axis=1)
+        res2 = df.drop(index=['a'], columns=['d'])
+        tm.assert_frame_equal(res1, res2)
+
```
Comment (jreback) [inline pandas/tests/frame/test_axis_select_reindex.py]: test with columns as well

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm some minor doc comments
prob should update usage in the main docs

File: pandas/core/generic.py [hunk 4]
```diff
@@ -2333,14 +2333,21 @@ def reindex_like(self, other, method=None, copy=True, limit=None,
...
         Return new object with labels in requested axis removed.
 
         Parameters
         ----------
         labels : single label or list-like
+            Index or column labels to drop.
         axis : int or axis name
+            Whether to drop labels from the index (0 / 'index') or
+            columns (1 / 'columns').
+        index, columns : single label or list-like
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: Perhaps add a `Notes` section saying that specifying both `labels` and `index` or `columns` will raise an error.

File: pandas/tests/frame/test_axis_select_reindex.py (hunk 3 above)
Comment (jorisvandenbossche) [reply pandas/tests/frame/test_axis_select_reindex.py]: Did you mean for the error test? (added one for that) Or for one of the tests above?

File: doc/source/whatsnew/v0.21.0.txt (hunk 1 above)
Comment (jorisvandenbossche) [reply doc/source/whatsnew/v0.21.0.txt]: this is only needed when you actually want to explicitly link to it from somewhere else in the docs, so I personally prefer to only add it when actually needed

Comment (jreback) [reply doc/source/whatsnew/v0.21.0.txt]: no, all sub-sections should have a reference as a general rule. yes we may not always link to it, but by definition when creating a non-trivial sub-section it should have a reference

File: pandas/core/generic.py [hunk 5]
```diff
@@ -2333,14 +2333,21 @@ def reindex_like(self, other, method=None, copy=True, limit=None,
...
+            Alternative to specifying `axis` (``labels, axis=1`` is
+            equivalent to ``columns=labels``).
```
Comment (jreback) [inline pandas/core/generic.py]: add a versionadded tag

File: pandas/tests/frame/test_axis_select_reindex.py (hunk 3 above)
Comment (jreback) [reply pandas/tests/frame/test_axis_select_reindex.py]: yep

**Review by jreback — APPROVED** [review]
minor comments. merge on green.

File: doc/source/whatsnew/v0.21.0.txt (hunk 1 above)
Comment (jorisvandenbossche) [reply doc/source/whatsnew/v0.21.0.txt]: There is nothing "by definition" about it (I mean, sphinx does not require it). IMO it just gives more overhead by always asking contributors to do this.

**Comment by TomAugspurger** [comment]
Thanks Joris!

**Pull request merged**